tasks/internal_reconcile_report_phantom/test/unit_tests/test_internal_reconcile_report_phantom.py

```diff
@@ -2,25 +2,284 @@
+                    0 : internal_reconcile_report_phantom.PAGE_SIZE
+                ],
+            },
+            result,
+        )
+
+    @patch("internal_reconcile_report_phantom.get_phantoms_sql")
+    def test_query_db_happy_path(self, mock_get_phantoms_sql: MagicMock):
+        """
+        Should query the db, then format the returned data.
+        """
+        job_id = random.randint(0, 999)  # nosec
+        page_index = Mock()
+
+        collection_id = uuid.uuid4().__str__()
+        granule_id = uuid.uuid4().__str__()
+        filename = uuid.uuid4().__str__()
+        key_path = uuid.uuid4().__str__()
+        orca_etag = uuid.uuid4().__str__()
+        orca_last_update = random.randint(0, 999999)  # nosec
+        orca_size = random.randint(0, 999)  # nosec
+
+        returned_row0 = {
+            "collection_id": collection_id,
+            "granule_id": granule_id,
+            "filename": filename,
+            "key_path": key_path,
+            "orca_etag": orca_etag,
+            "orca_last_update": orca_last_update,
+            "orca_size": orca_size,
+        }
+        mock_execute = Mock(return_value=[returned_row0])
+        mock_connection = Mock()
+        mock_connection.execute = mock_execute
+        mock_exit = Mock(return_value=False)
+        mock_enter = Mock()
+        mock_enter.__enter__ = Mock(return_value=mock_connection)
+        mock_enter.__exit__ = mock_exit
+        mock_engine = Mock()
+        mock_engine.begin = Mock(return_value=mock_enter)
+
+        result = internal_reconcile_report_phantom.query_db(
+            mock_engine,
+            job_id,
+            page_index,
+        )
 
-    def test_dummy():
-        pass
+        mock_enter.__enter__.assert_called_once_with()
+        mock_execute.assert_called_once_with(
+            mock_get_phantoms_sql.return_value,
+            [
+                {
+                    "job_id": job_id,
+                    "page_index": page_index,
+                    "page_size": internal_reconcile_report_phantom.PAGE_SIZE,
+                }
+            ],
+        )
+        mock_exit.assert_called_once_with(None, None, None)
+        mock_get_phantoms_sql.assert_called_once_with()
+        self.assertEqual(
+            [
+                {
+                    internal_reconcile_report_phantom.PHANTOMS_COLLECTION_ID_KEY: collection_id,
+                    internal_reconcile_report_phantom.PHANTOMS_GRANULE_ID_KEY: granule_id,
+                    internal_reconcile_report_phantom.PHANTOMS_FILENAME_KEY: filename,
+                    internal_reconcile_report_phantom.PHANTOMS_KEY_PATH_KEY: key_path,
+                    internal_reconcile_report_phantom.PHANTOMS_ORCA_ETAG_KEY: orca_etag,
+                    internal_reconcile_report_phantom.PHANTOMS_ORCA_LAST_UPDATE_KEY: orca_last_update,
+                    internal_reconcile_report_phantom.PHANTOMS_ORCA_SIZE_KEY: orca_size,
+                }
+            ],
+            result,
+        )
```

Comment:
I am wondering if we shouldn't have another "happy" test where in the database returns an empty data set (no phantom records for the given job_id). Thoughts? Does the output schema still hold up if phantoms is an empty array?

Reply:
It is valid. There is no provision for how many items are returned.